src/org/traccar/protocol/Tk103ProtocolDecoder.java

```diff
@@ -63,6 +66,10 @@ public Tk103ProtocolDecoder(Tk103Protocol protocol) {
             .number("(?:L(x+))?")                // odometer
             .any()
             .number("([+-]ddd.d)?")              // temperature
+            .groupBegin()
+            .number("([+-]?d+.d{1,2}),").optional() // altitude
+            .number("(d+)$").optional()         // number of visible satellites
+            .groupEnd("?")
```

Comment:
Why do you have this group when both items in it are already optional?

Reply:
If they are present together, then you should remove `optional` from parameters themselves.

Reply:
done